sentry-android-core/src/main/java/io/sentry/android/core/InternalSentrySdk.java

```diff
@@ -0,0 +1,202 @@
+package io.sentry.android.core;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import io.sentry.DateUtils;
+import io.sentry.HubAdapter;
+import io.sentry.IHub;
+import io.sentry.ILogger;
+import io.sentry.ISerializer;
+import io.sentry.ObjectWriter;
+import io.sentry.Scope;
+import io.sentry.SentryEnvelope;
+import io.sentry.SentryEnvelopeItem;
+import io.sentry.SentryEvent;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.Session;
+import io.sentry.protocol.App;
+import io.sentry.protocol.Device;
+import io.sentry.protocol.SentryId;
+import io.sentry.protocol.User;
+import io.sentry.util.MapObjectWriter;
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Sentry SDK internal API methods meant for being used by the Sentry Hybrid SDKs. */
+@ApiStatus.Internal
+public final class InternalSentrySdk {
+
+  /**
+   * @return a copy of the current hub's topmost scope, or null in case the hub is disabled
+   */
+  @Nullable
+  public static Scope getCurrentScope() {
+    final @NotNull AtomicReference<Scope> scopeRef = new AtomicReference<>();
+    HubAdapter.getInstance().withScope(scopeRef::set);
```

Comment:
@markushi See https://github.com/getsentry/sentry-java/blob/beba01476ab71745b0cae821bed705d582374161/sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java#L46-L47
This has an issue with some specific use cases, please avoid method references.

Reply:
Also check if the source code has any more references to `::`.

Reply:
Are you sure this is a problem? I see more than 50 occurrences in our code. Most of them are backend-related, but there are also some which do get executed on Android devices (e.g. `JsonObjectDeserializer`)

Reply:
yes, I've written that comment myself, its a problem on Android only during desugaring, mainly because of older AGP versions (Hi RN and Flutter).
If there are more on Android code, they should be removed, that's why this comment https://github.com/getsentry/sentry-java/pull/2814#discussion_r1266363020

Reply:
@markushi can you raise a new issue for replacing all `::` references that run on Android code?

Reply:
`JsonObjectDeserializer` has references to `::` not sure if more.

Reply:
:+1: created an issue here, https://github.com/getsentry/sentry-java/issues/2848

Reply:
Following this logic, we should also get rid of all `default` methods in interfaces, because this is also part of the desugaring. We use those quite heavily. The fact, that nobody has reported any problems so far kinda confirms that there's noone of our customers on AGP lower than 3.0. Is still worth considering this limitation?
![image](https://github.com/getsentry/sentry-java/assets/4999776/be554645-7b26-455f-9c9e-ae6dfd5d0afd)

Reply:
@romtsn no, we decided and use Java 8 features, the min. version of AGP isn't a problem here, the method reference was a bug o the AGP desugaring and not "not supported" if not version >= X.
If `default` has a bug in one of the AGP versions that are above the min. required, then yes, we'd get rid of `default` methods for compatibility, which is not the case.

Reply:
Ok, what was the version where the bug has been fixed? If it's 3.0.1 I'd say we are still good to not consider it

Reply:
Don't recall, but why do we even bother if we can keep 100% compatibility by having just a few chars more? `() -> {}`